flasgger/base.py

```diff
@@ -433,20 +433,53 @@ def get_apispecs(self, endpoint='apispec_1'):
             openapi_version=openapi_version,
             doc_dir=self.config.get('doc_dir'))
 
+        def merge_sub_component(dest, key, source):
+            if len(source) > 0 and dest.get(key) is None:
+                dest[key] = {}
+            if len(source) > 0 and len(dest[key]) >= 0:
+                dest[key].update(source)
+
         http_methods = ['get', 'post', 'put', 'delete']
         for rule, verbs in specs:
             operations = dict()
             for verb, swag in verbs:
 
                 if is_openapi3(openapi_version):
-                    update_dict = swag.get('components', {}).get('schemas', {})
+                    source_components = swag.get('components', {})
+                    update_schemas = source_components.get('schemas', {})
+                    merge_sub_component(data['components'], 'parameters',
```

Comment:
could these all be in a for-loop, for a list of the new keys?

Reply:
Good call, let me add that right now.